sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -549,15 +549,18 @@ class HDFSFileCatalog(
 
   refresh()
 
+  private def isInvalidFile(f: FileStatus): Boolean =
+    f.getPath.getName.startsWith("_") || f.getLen == 0
```

Comment:
Actually there is an existing [CSV test case][1] that cover empty CSV input files. I have to [remove the empty file assertion][2] in `FileSourceStrategy` for this test case when implementing `buildReader()` for CSV.

[1]: https://github.com/apache/spark/blob/e5a1b301fbe191f1a9627a1083d960c98f543d13/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala#L248-L256
[2]: https://github.com/apache/spark/pull/12002#discussion_r57594679